Mathlib/Order/SuccPred/LinearLocallyFinite.lean

```diff
@@ -428,3 +437,9 @@ def orderIsoRangeOfLinearSuccPredArch [OrderBot ι] [OrderTop ι] :
     rw [← @toZ_le_iff ι _ _ _ _ ⊥, Int.toNat_of_nonneg (toZ_nonneg bot_le)]
 
 end OrderIso
+
+instance (priority := 100) Countable.of_linearOrder_locallyFiniteOrder [LocallyFiniteOrder ι] :
+    Countable ι :=
+  have := LinearLocallyFiniteOrder.succOrder ι
+  have := LinearLocallyFiniteOrder.predOrder ι
```

Comment:
Since `succOrder` and `predOrder` contain data, shouldn't these be `let` (or `letI`) rather than `have`?

Reply:
If the proof works, I think you are free to use `have`, which might reduce some burden for Lean.